### PR TITLE
Introduce Playwright plugin config

### PR DESCRIPTION
**Pull request by Tlacenka**
In this PR I introduce ESLint config for Playwright using their [official plugin](https://github.com/playwright-community/eslint-plugin-playwright).

## Notable changes
- While the plugin is primarily aimed at E2E projects, I also included the regular `spec`|`test` pattern.
- For now I created just one configuration. Modifications might be needed for projects heavily utilising POMs.
- I reduced severity for conditionals based on similar approach in the `jest` config.

### CODE REVIEW

File: src/configs/playwright.js
Comment (matejchalk) [inline src/configs/playwright.js:1]: How about adding [`eslint-plugin-jest-formatting`](https://www.npmjs.com/package/eslint-plugin-jest-formatting) to ensure padding lines around `it`/`test` blocks? Don't let the name confuse you, it works for any Jest-like syntax, it's already being used in Vitest config.

**Review by matejchalk — COMMENTED** [review]
Thank you so much for this contribution :tada: :rocket:

File: package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "eslint-plugin-jest": "^28.8.0",
     "eslint-plugin-jest-formatting": "^3.0.0",
     "eslint-plugin-n": ">=17.0.0",
+    "eslint-plugin-playwright": ">=2.1.0",
```
[the remark this answers is not in the document]
Comment (Tlacenka) [reply package.json:53]: Didn't know that, thank you!

File: docs/playwright.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+| :-----------------------------------------------------------------------------------------------------------------------: | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :------ | :-----: | :-------: |
+| [![playwright](./icons/material/playwright.png)](https://github.com/playwright-community/eslint-plugin-playwright#readme) | [missing-playwright-await](https://github.com/playwright-community/eslint-plugin-playwright/tree/main/docs/rules/missing-playwright-await.md)<br />Identify false positives when async Playwright APIs are not properly awaited. |         |   🔧    |           |
+| [![playwright](./icons/material/playwright.png)](https://github.com/playwright-community/eslint-plugin-playwright#readme) | [no-focused-test](https://github.com/playwright-community/eslint-plugin-playwright/tree/main/docs/rules/no-focused-test.md)<br />Prevent usage of `.only()` focus test annotation                                                |         |   💡    |           |
```
[the remark this answers is not in the document]
Comment (Tlacenka) [reply docs/playwright.md:32]: Agreed, fixed.

File: docs/playwright.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+| [![playwright](./icons/material/playwright.png)](https://github.com/playwright-community/eslint-plugin-playwright#readme) | [no-networkidle](https://github.com/playwright-community/eslint-plugin-playwright/tree/main/docs/rules/no-networkidle.md)<br />Prevent usage of the networkidle option                                                           |         |         |           |
+| [![playwright](./icons/material/playwright.png)](https://github.com/playwright-community/eslint-plugin-playwright#readme) | [no-standalone-expect](https://github.com/playwright-community/eslint-plugin-playwright/tree/main/docs/rules/no-standalone-expect.md)<br />Disallow using `expect` outside of `test` blocks                                      |         |   🔧    |           |
```
[the remark this answers is not in the document]
Comment (Tlacenka) [reply docs/playwright.md:34]: I suppose... the test will still technically pass. Changed to warning.

File: src/lib/patterns.js [hunk 4]
```diff
@@ -72,6 +72,12 @@ export const CYPRESS_FILE_PATTERNS = [
   '**/e2e/**/*.[jt]s?(x)',
 ];
 
+export const PLAYWRIGHT_FILE_PATTERNS = [
+  '**/*.+(test|spec).[jt]s?(x)',
+  '**/*.po.[jt]s?(x)',
+  '**/e2e/**/*.[jt]s?(x)',
```
[the remark this answers is not in the document]
Comment (Tlacenka) [reply src/lib/patterns.js:78]: Good point, added.

File: src/configs/playwright.js
Comment (Tlacenka) [reply src/configs/playwright.js:1]: Sounds good, added.

**Review by matejchalk — APPROVED** [review]

**Pull request merged**